src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -224,6 +241,11 @@ internal void MakeReadOnly()
             _isReadOnly = true;
         }
 
+        internal ServiceAuthorizationBehavior Clone()
```

Comment:
Make this class implement ICloneable and then ServiceDescription can be general cased to if it implements ICloneable, case to that and call Clone.  Only differences is ICloneable returns `object` so you would need to cast, but that's not a big deal.

Reply:
done. It looks better this way. Thanks!